### PR TITLE
Feature/datatransfer api gha tests

### PR DESCRIPTION
**Pull request by davidjovanovic**

### CODE REVIEW

File: data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java [hunk 1]
```diff
@@ -96,7 +97,7 @@ public TransferProcess initiateDataTransfer(TransferRequestMessage transferReque
 				.format(transferRequestMessage.getFormat())
 				.dataAddress(transferRequestMessage.getDataAddress())
 				.state(TransferState.REQUESTED)
-				.role(transferProcess.getRole())
+				.role(IConstants.ROLE_PROVIDER)
```
Comment (IgorBalog-Eng) [inline data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java:100]: This one is not correctly set when creating TP.initialized?

File: data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java [hunk 2]
```diff
@@ -158,8 +159,16 @@ public TransferProcess startDataTransfer(TransferStartMessage transferStartMessa
 		log.debug("Starting data transfer for consumerPid {} and providerPid {}", consumerPidFinal, providerPidFinal);
 
 		TransferProcess transferProcessRequested = findTransferProcess(consumerPidFinal, providerPidFinal);
+		
+		if (IConstants.ROLE_PROVIDER.equals(transferProcessRequested.getRole()) && TransferState.REQUESTED.equals(transferProcessRequested.getState())) {
```
Comment (IgorBalog-Eng) [inline data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java:163]: Do we have test cases for this logic?

**Review by IgorBalog-Eng — COMMENTED** [review]
Review

File: data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java (hunk 1 above)
Comment (davidjovanovic) [reply data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java:100]: It was, but some tests could have been done with some workaround and passed. With this we are assuring that only ones which make sense can be written in a proper maner.

File: data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java (hunk 2 above)
Comment (davidjovanovic) [reply data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java:163]: Yep, they were already there. I had to update them to work with the new changes

**Review by marest94 — APPROVED** [review]

**Review by IgorBalog-Eng — APPROVED** [review]
One log missed but approve is here

**Pull request merged**